src/ape/pytest/fixtures.py

```diff
@@ -20,3 +24,78 @@ def chain(self) -> ChainManager:
     @pytest.fixture(scope="session")
     def project(self) -> ProjectManager:
         return self.project_manager
+
+    @pytest.fixture(scope="session")
+    def _session_isolation(self) -> Iterator[None]:
+        snapshot_id = None
+        try:
+            snapshot_id = self.chain_manager.snapshot()
+        except NotImplementedError:
+            self._warn_for_unimplemented_snapshot()
+
+        yield
+
+        if snapshot_id is not None and snapshot_id in self.chain_manager._snapshots:
+            self.chain_manager.restore(snapshot_id)
+
+    @pytest.fixture(scope="package")
+    def _package_isolation(self, _session_isolation) -> Iterator[None]:
```

Comment:
Does this need to be called "_session_isolation" to work? (Looking at the logic in the runners for dynamically setting the fixture)

Reply:
ah the leading underscore you mean? That's just so it's considered internal, and not shown to the end user via `pytest --fixtures`